src/utils/numbers.ts

```diff
@@ -0,0 +1,26 @@
+export interface FactorMap {
+    divisor: number,
+    labels: string[],
+}
+
+export const bytesMap: FactorMap = {
+    divisor: 1024,
+    labels: [' Bytes', ' KiB', ' MiB', ' GiB', ' TiB', ' PiB', ' EiB', ' ZiB', ' YiB']
+}
+
+export const kiloMap: FactorMap = {
+    divisor: 1000,
+    labels: ['', 'k', 'M']
+}
+
+export function getFactor(value: number, map:FactorMap) {
```

Comment:
I think this one really deserves a JSDoc with `@return` tag: Just by looking at it one will not quickly be able to understand whether
`value: 12345678, map: kiloMap` will return `1, ''`, `1000, 'k'` or `1000000, 'M'` - and use cases for all three outputs definitely exist.

E.g. at first look, I presumed that `1000` would become `1 k` but `1234` would remain `1234` - but this is not what happens, it is merely another, no less reasonable expectation.

Reply:
Please run the tests to see and understand whats happing. `kiloMap` and `bytesMap` are working different.

Reply:
Well, I did. And I even checked the math (for whatever that's worth - very rusty). The code is perfectly fine. For in-house code, the whole function is perfectly fine.

I still do think though, that for a library it is a bit... unbecoming to expect the user to glean its capabilities from code and tests (and, yes, a single example showing that it works at all) alone. Imagine having to do that with Spring!
(Myself, I had to, several times, debug various parts of the Spring framework, even though I was a pure consumer, just to find contractual information omitted from the documentation - really makes one consider dropping the library. Or, as the old adage goes: "Code is easier written than read.")

So, I think it would be nice to have some basic contract specific to this function right _on_ the function (and thence available as mouse-over in IDEs such as Eclipse or IntelliJ).

With my earlier comment closed that mayhap negative values should also be considered, I'd add not just the `@return` but also a `@param value must be positive` - the limitation is fine, but it should really be spelled out (or even better: expressed through the type system; but that is not viable for `value` here).
A `@param map must be sufficiently filled` might also be sensible. Again, I consider the _behaviour_ perfectly fine, but giving a heads-up would be "the Gentleman's way".
FWIW: I'd write into the `@return` something alone the lines of "representation of `value` with the decimal point moved behind the most significant digit, plus the matching label from `map`, if any".

Reply:
I've updated the documentation.